backend/medtagger/repositories/roles.py

```diff
@@ -0,0 +1,35 @@
+"""Module responsible for definition of Roles' Repository."""
+from typing import Optional
+
+from medtagger.api import InvalidArgumentsException
+from medtagger.database import db_session
+from medtagger.database.models import Role
+from medtagger.repositories.users import UsersRepository
+
+
+class RolesRepository(object):
+    """Repository for Roles."""
+
+    @staticmethod
+    def get_role_with_name(role_name: str) -> Optional[Role]:
+        """Get role with given name.
+
+        :return Optional of role
+        """
+        with db_session() as session:
+            role = session.query(Role).filter(Role.name == role_name).first()
+        return role
+
+    @staticmethod
+    def set_user_role(user_id: int, role_name: str) -> None:
+        """Set user's role. Old role is being replaced."""
+        user = UsersRepository.get_user_by_id(user_id)
+        if user is None:
```

Comment:
Please change:
```python
user = UsersRepository.get_user_by_id(user_id)
if user is None:
    raise InvalidArgumentsException('User with this id does not exist.')
```

...to:
```python
try:
    user = UsersRepository.get_user_by_id(user_id)
except NotFoundError:
    raise InvalidArgumentsException('User with this id does not exist.')
```

But at first, please check what exception is raised by SQLAlchemy's `.one()` method in case the entry is missing.

More about "Asking for permission instead of forgiveness":
https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html

Reply:
Fixed (Changed to `.one()` in user and role repositories, and FYI SQLAlchemy returns `NoResultFound` if query select no rows.)